src/Magnum/Platform/GlfwApplication.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016
+              Vladimír Vondruš <mosra@centrum.cz>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "GlfwApplication.h"
+
+#include <tuple>
+
+#include "Magnum/Version.h"
+#include "Magnum/Platform/Context.h"
+#include "Magnum/Platform/ScreenedApplication.hpp"
+
+namespace Magnum { namespace Platform {
+
+typedef GlfwApplication::Configuration::Flag Flag;
+typedef GlfwApplication::Configuration::Flags Flags;
+typedef GlfwApplication::Configuration::WindowFlag WindowFlag;
+typedef GlfwApplication::Configuration::WindowFlags WindowFlags;
```

Comment:
Hmm... I see that the flags are used only in the `tryCreateContext()`, right? Then I would move these typedefs to the function block so they don't leak outside.

Also, maybe after the simplification the additional `Configuration::` prefix wouldn't do *that* much harm and so these typedefs won't be needed at all.